src/wp-includes/class-wp-meta-manager.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * A registry for meta data.
+ *
+ * @package WordPress
+ * @since 4.6.0
+ */
+final class WP_Meta_Manager {
+	/**
+	 * Registered meta of the post meta type (including pages and CPTs).
+	 *
+	 * @since 4.6.0
+	 * @access private
+	 * @var array
+	 */
+	private $post = array(
+		'all' => array()
+	);
+
+	/**
+	 * Registered meta of the taxonomy meta type.
+	 *
+	 * @since 4.6.0
+	 * @access private
+	 * @var array
+	 */
+	private $taxonomy = array(
+		'all' => array()
+	);
+
+	/**
+	 * Registered meta of the user meta type.
+	 *
+	 * @since 4.6.0
+	 * @access private
+	 * @var array
+	 */
+	private $user = array(
+		'all' => array()
+	);
+
+	/**
+	 * Registered meta of the comment meta type.
+	 *
+	 * @since 4.6.0
+	 * @access private
+	 * @var array
+	 */
+	private $comment = array(
+		'all' => array()
+	);
+
+	/**
+	 * @since 4.6.0
+	 */
+	function __construct() {
+		$post_types_without_meta = array( 'revision', 'nav_menu_item' );
+		foreach ( get_post_types( null, 'names' ) as $post_type ) {
+			if ( in_array( $post_type, $post_types_without_meta ) ) {
+				continue;
+			}
+			$this->post[$post_type] = array();
+		}
+
+		foreach ( get_taxonomies() as $taxonomy ) {
+			$this->taxonomy[$taxonomy] = array();
+		}
+
+		foreach ( array( 'comment', 'ping' ) as $comment_type ) {
+			$this->comment[$comment_type] = array();
+		}
+	}
+
+	/**
+	 * Register a meta key.
+	 *
+	 * @since 4.6.0
+	 *
+	 * @param array $query {
+	 *     Array of meta data properties.
+	 *
+	 *     @type string $object_type  The meta type this key belongs to, e.g. `post`.
+	 *     @type string $object_subtype The meta subtype this key belongs to, e.g. `page`.
+	 *     @type string $key          The meta key.
+	 *     @type array  $schema       The meta field's schema, according to JSON schema.
+	 *     @type array  $public       Whether the meta key should be publicly queryable.
+	 * @return true|WP_Error
+	 */
+	public function register( $args ) {
+		if ( empty( $args['object_type'] ) ) {
```

Comment:
Sure, sounds good. I believe Jeremy had similar thoughts in inline comments in `register_meta()`, to match similar casting in `register_post_type()` and elsewhere.